docs/source/configurator.rst

```diff
@@ -0,0 +1,63 @@
+Configuration files and ``ebu-run``
+===================================
+
+There are 2 ways the nodes and carriage mechanisms can be set up:
+
+    1. Write a script where you instantiate these components and connect them. This technique is making use of
+    dependency injection. This means the nodes and carriage mechanisms do not need to know about the creation of
+    each other. They just get the already created component referenced as a parameter, and check if that component
+    implement the expected interface. This eliminates the possibility of creating spaghetti code where a node would know
+    how to create a carriage mechanism and deal with all the possible parameters and a carriage mechanism would know
+    in turn how to instantiate a node. This is to be avoided. And this brings us to the second way to do this.
+
```

Comment:
Might be worth referencing that design pattern documentation in our documentation?